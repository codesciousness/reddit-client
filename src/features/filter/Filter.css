.Filter__button {
    color: white;
    font-size: 1rem;
    font-weight: bold;
    background: transparent;
    border: 1px solid white;
    border-radius: 50px;
    padding: 0.75rem 2rem;
    margin: 0.5rem;
    cursor: pointer;
    opacity: 0.8;
    transition: all 0.3s ease-in-out;
}

.Filter__button:hover {
    color: black;
    background: white;
}

.Filter__button__active {
    color: black;
    font-size: 1rem;
    font-weight: bold;
    background: white;
    border: 1px solid white;
    border-radius: 50px;
    padding: 0.75rem 2rem;
    margin: 0.5rem;
    cursor: pointer;
    opacity: 0.8;
    transition: all 0.3s ease-in-out;
}