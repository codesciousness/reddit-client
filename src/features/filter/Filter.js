import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import './Filter.css';
import { setFilter, clearFilter, selectFilter } from './filterSlice';

const Filter = ({ flair }) => {
    let filter = useSelector(selectFilter);
    const dispatch = useDispatch();
    const flairId = flair.replace(' ', '');
    const handleClick = ({ target }) => {
        if (filter === flair) {
            dispatch(clearFilter());
        }
        else {
            dispatch(setFilter(target.id));
        }
    }
    return (
        <button id={flairId} className={filter === flair ? "Filter__button__active" : "Filter__button"} onClick={handleClick}>{flair}</button>
    );
}

export default Filter;