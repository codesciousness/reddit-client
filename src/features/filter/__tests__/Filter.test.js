import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Filter from '../Filter';

const flair = 'Chemistry';

const mockStore = configureStore([]);

describe('<Filter />', () => {
    let store;

    beforeEach(() => {
        store = mockStore({
            filter: ''
        });
    });

    test('Filter component is defined', () => {
        expect(<Filter />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(
            <Provider store={store}>
                <Filter flair={flair} />
            </Provider>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders inactive button correctly with props', () => {
        const { container } = render(
            <Provider store={store}>
                <Filter flair={flair} />
            </Provider>
        );
        expect(container.getElementsByClassName('Filter__button')[0]).toBeInTheDocument();
        expect(screen.getByText('Chemistry')).toBeInTheDocument();
    });
    test('renders active button correctly with props', () => {
        store = mockStore({
            filter: 'Chemistry'
        });
        const { container } = render(
            <Provider store={store}>
                <Filter flair={flair} />
            </Provider>
        );
        expect(container.getElementsByClassName('Filter__button__active')[0]).toBeInTheDocument();
        expect(screen.getByText('Chemistry')).toBeInTheDocument();
    });
    test('dispatches correct action when inactive filter button is clicked', () => {
        render(
            <Provider store={store}>
                <Filter flair={flair} />
            </Provider>
        );
        const button = screen.getByText('Chemistry');
        userEvent.click(button);
        const actions = store.getActions();
        const expectedPayload = {type: 'filter/setFilter', payload: 'Chemistry'};
        expect(actions).toEqual([expectedPayload]);
    });
    test('dispatches correct action when active filter button is clicked', () => {
        store = mockStore({
            filter: 'Chemistry'
        });
        render(
            <Provider store={store}>
                <Filter flair={flair} />
            </Provider>
        );
        const button = screen.getByText('Chemistry');
        userEvent.click(button);
        const actions = store.getActions();
        const expectedAction = {type: 'filter/clearFilter'};
        expect(actions).toEqual([expectedAction]);
    });
});