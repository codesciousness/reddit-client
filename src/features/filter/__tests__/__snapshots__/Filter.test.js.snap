// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> matches the snapshot 1`] = `
<button
  className="Filter__button"
  id="Chemistry"
  onClick={[Function]}
>
  Chemistry
</button>
`;
